PusherServer/IPusher.cs

```diff
@@ -60,6 +60,8 @@ public interface IPusher
         /// For any non-200 response from the HTTP API or if the response body cannot be parsed as JSON
         /// </exception>
         ITriggerResult Trigger(string[] channelNames, string eventName, object data, ITriggerOptions options);
+
+        ITriggerResult Trigger(BatchEvent[] events);
```

Comment:
Using the low-level Array is nice enough here IMO. It's easy for a developer to use a more complex data type and then convert it to an Array when calling `Trigger`.